### PR TITLE
Single error type

### PR DESCRIPTION
**Pull request by nlordell**
This PR introduces a common `RejectionReason` type that implements `Reject` and `Into<Rejection>` so that it can be used with the usual Rust `?` sugar, allowing for differentiation between rejection reasons without having to create a new type for each one.

### Test Plan

Run the price estimator and check that the rejections still work:
```
$ cargo run -p price-estimator
...
$ curl -s 'http://localhost:8080/api/v1/markets/9-7/estimated-buy-amount/1?atoms=false' | jq
{
  "message": "requested base units for token with missing ERC20 info"
}
$ curl -s 'http://localhost:8080/api/v1/markets/9-DAI/estimated-buy-amount/1?atoms=false' | jq
{
  "message": "token symbol or address not found"
}
```

### CODE REVIEW

File: price-estimator/src/filter.rs [hunk 1]
```diff
@@ -24,40 +20,32 @@ pub fn all(
...
-        message = "request with atoms=true for token we don't have erc20 info for";
-    } else if let Some(InternalError(err)) = err.find() {
-        log::warn!("internal server error: {:?}", err);
-        code = StatusCode::INTERNAL_SERVER_ERROR;
-        message = "internal server error";
+    let (code, message) = if let Some(reason) = err.find::<RejectionReason>() {
+        log::warn!("rejection reason: {:?}", reason);
+        reason.as_http_error()
+    } else if err.is_not_found() {
+        (StatusCode::NOT_FOUND, "invalid url path")
```
Comment (fleupold) [inline price-estimator/src/filter.rs:39]: Can we log the requests here and in the BAD_REQUEST case as well? I was seeing some 404 in the logs but wasn't able to find out the exact qeuery that did them.

**Review by fleupold — COMMENTED** [review]
Looks like this PR got caught up with the CurrencyPair change. Can you rebase?

**Comment by nlordell** [comment]
> Can you rebase?

Done!

File: price-estimator/src/filter.rs (hunk 1 above)
Comment (nlordell) [reply price-estimator/src/filter.rs:39]: So I think the warp log filter should log all requests currently. Its probably a logging configuration issue in the deployment.

Comment (nlordell) [reply price-estimator/src/filter.rs:39]: Addressed by #1302 and #1303

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2F1RkDDoIVs3ntm%2Fgiphy.gif&token=error" data-gitmeme-token="error"></a>

**Review by e00E — APPROVED** [review]

**Pull request merged**